code/game/machinery/electrolyzer.dm

```diff
@@ -0,0 +1,183 @@
+#define ELECTROLYZER_MODE_STANDBY	"standby"
+#define ELECTROLYZER_MODE_WORKING	"working"
+
+/obj/machinery/electrolyzer
+	anchored = FALSE
+	density = TRUE
+	interaction_flags_machine = INTERACT_MACHINE_ALLOW_SILICON | INTERACT_MACHINE_WIRES_IF_OPEN
+	icon = 'icons/obj/atmos.dmi'
+	icon_state = "electrolyzer-off"
+	name = "space electrolyzer"
+	desc = "Thanks to the fast and dynamic response of our electrolyzers, on-site hydrogen production is guaranteed. Warranty void if used by clowns"
+	max_integrity = 250
+	armor = list("melee" = 0, "bullet" = 0, "laser" = 0, "energy" = 0, "bomb" = 0, "bio" = 100, "rad" = 100, "fire" = 80, "acid" = 10)
+	circuit = /obj/item/circuitboard/machine/electrolyzer
+	ui_x = 400
+	ui_y = 305
+
+	///used to check if there is a cell in the machine
+	var/obj/item/stock_parts/cell/cell
+	///check if the machine is on or off
+	var/on = FALSE
+	///check what mode the machine should be (WORKING, STANDBY)
+	var/mode = ELECTROLYZER_MODE_STANDBY
+	///Increase the amount of moles worked on, changed by upgrading the laser tier
+	var/workingPower = 1
+	///Decrease the amount of power usage, changed by upgrading the capacitor tier
+	var/efficiency = 0.5
+
+/obj/machinery/electrolyzer/get_cell()
+	return cell
+
+/obj/machinery/electrolyzer/Initialize()
+	. = ..()
+	cell = new(src)
+	update_icon()
+
+/obj/machinery/electrolyzer/on_construction()
+	QDEL_NULL(cell)
+	panel_open = TRUE
+	update_icon()
+	return ..()
+
+/obj/machinery/electrolyzer/on_deconstruction()
+	if(cell)
+		component_parts += cell
+		cell = null
```

Comment:
Will this variable be properly cleared if an admin right-click deletes the machine?

Reply:
i think? if not how can i fix it?

Reply:
You could add a check on `Destroy()`
Start by finding out if it's an issue, and fix it if it's so.

Reply:
deleting the machine by admin command will remove the machine but will release the cell on the ground. This is used to ensure that the player receive the correct cell that has been inserted when the machine gets deconstructed.